unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Interfaces/EntityIdHelper.cs

```diff
@@ -35,12 +39,57 @@ public long EntityFromLegacyEntityString(string entityId)
                     return (long) SpecialEntityId.THIRD_PERSON_CAMERA_ENTITY_REFERENCE;
             }
 
-            long entityIdLong = entityId.GetHashCode() << 9;
-
+            long entityIdLong = DecodeBase36(entityId) << 9;
+            
             if (!entityIdToLegacyId.ContainsKey(entityIdLong))
                 entityIdToLegacyId[entityIdLong] = entityId;
 
             return entityIdLong;
         }
+        
+        public Int64 DecodeBase36(string input)
+        {
+            string charList = "0123456789abcdefghijklmnopqrstuvwxyz";
```

Comment:
Make this a `const` to avoid allocation. You can do it here, it doesn't need to go on top of the class.